internal/updater/client.go

```
@@ -0,0 +1,111 @@
+	}
+}
+
+// GetLatestRelease fetches the latest release from GitHub
+func (c *Client) GetLatestRelease() (*GitHubRelease, error) {
+	url := fmt.Sprintf("https://api.github.com/repos/%s/%s/releases/latest", c.owner, c.repo)
+
+	resp, err := c.httpClient.Get(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch latest release: %w", err)
+	}
+	defer resp.Body.Close()
```

golangci-lint (1.64.8)

Error return value of `resp.Body.Close` is not checked

(errcheck)